packages/flutter/lib/src/material/list_item.dart

```diff
@@ -36,11 +37,17 @@ class ListItem extends StatelessComponent {
   final Widget right;
   final bool isThreeLine;
   final bool isDense;
```

Comment:
Just a thought, not a comment on this patch: maybe we should convert isThreeLine and isDense into a single enum that gives the various ways that this can be laid out.